src/components/user-steps-wrapper/UserStepsWrapper.tsx

```diff
@@ -26,17 +27,27 @@ const UserStepsWrapper: FC<UserStepsWrapperProps> = ({ userRole }) => {
   const { setNeedConfirmation } = useContext(ConfirmationDialogContext)
   setNeedConfirmation(true)
 
+  useEffect(() => {
+    setNeedConfirmation(true)
```

Comment:
It looks like after adding this line, we do not need to call `setNeedConfirmation` on line 28.